lib/node/runner/api/tx_operations.go

```diff
@@ -31,17 +33,51 @@ func (api NetworkHandlerAPI) GetOperationsByTxHashHandler(w http.ResponseWriter,
 	options := p.ListOptions()
 
 	if httputils.IsEventStream(r) {
-		event := fmt.Sprintf("txhash-%s", hash)
-		es := NewEventStream(w, r, renderEventStream, DefaultContentType)
-		ops, _ := api.getOperationsByTxHash(hash, options)
-		for _, op := range ops {
-			es.Render(op)
+		var blk *block.Block
+		es := NewEventStream(
+			w,
+			r,
+			func(args ...interface{}) ([]byte, error) {
+				if len(args) <= 1 {
+					return nil, fmt.Errorf("render: value is empty")
+				}
+				i := args[1]
+
+				if i == nil {
+					return nil, nil
+				}
+
+				if blk == nil {
+					if blk, err = api.getBlockByTxHash(hash); err != nil {
+						return nil, err
+					}
+				}
+
+				r := resource.NewOperation(i.(*block.BlockOperation))
```

Comment:
What do you think that type checking about `i.(*blocks.BlockOperation)` ?

Reply:
@anarcher `i` is `interface{}`.

Reply:
```suggestion
                             bo,ok := i.(*block.BlockOperation)
                             if !ok {
                                 return nil,nil
                             }
				r := resource.NewOperation(bo)
```

Reply:
But, Observer.Trigger do guarantee i as `*block.BlockOperation`. This checking is not needed.

Reply:
@anarcher In this block, `renderFunc` should be get `*block.BlockOperation`, so as you suggested checking thru `ok` is unnecessary. Without type assertion, `i.(*block.BlockOperation)` it causes,
```
cannot use i (type interface {}) as type *block.BlockOperation in argument to resource.NewOperation: need type assertion
```

Reply:
`t := i.(T)` :  If i does not hold a T, the statement will trigger a panic.

`t, ok := i.(T)` :  If not, ok will be false and t will be the zero value of type T, and no panic occurs.

My suggestion is  `t,ok:=i.(T)`. Plz correct me :->